audio/pcm_decode.c

```diff
@@ -1087,6 +1090,7 @@ static int pcm_enqueuebuffer(FAR struct audio_lowerhalf_s *dev,
     {
       /* Parse and verify the candidate PCM WAV file header */
 
+#ifndef CONFIG_AUDIO_FORMAT_RAW
       if (pcm_parsewav(priv, &apb->samp[apb->curbyte]))
```

Comment:
```suggestion
      if (bytesleft >= sizeof(struct wav_header_s) && pcm_parsewav(priv, &apb->samp[apb->curbyte]))
```
or
```suggestion
      if (pcm_parsewav(priv, &apb->samp[apb->curbyte], bytesleft))
```
And remove the check at line 1089

Reply:
@xiaoxiang781216 In case of  CONFIG_AUDIO_FORMAT_RAW pcm_parsewav is not defined and not used. Please merge the patch as is

Reply:
you can select the second form: pass the length to pcm_parsewav and let pcm_parsewav verify the length for us. The current implementation will report the false alarm when the raw pcm file length is smaller than struct wav_header_s.

Reply:
done. please accept the ifdef in place